### PR TITLE
refactor: Move "about" formatting logic into dedicated classes

### PR DESCRIPTION
**Pull request by edgarrmondragon**
----
:books: Documentation preview :books:: https://meltano-sdk--1570.org.readthedocs.build/en/1570/

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
@aaronsteers We've talked about this before and with the recent discussions around `--about` output for `oneOf` configurations I decided to take a stab at refactoring the _about_ formatters.

The refactored code is much easier to test, especially with snapshots. Let me know that you think.

**Comment by aaronsteers** [comment]
This is cool. Can we discuss again in office hours before merging?

Just want to make sure I understand the changes and that we collect feedback on any interesting implemention choices. Does that work?

**Comment by edgarrmondragon** [comment]
> Just want to make sure I understand the changes and that we collect feedback on any interesting implemention choices. Does that work?

@aaronsteers There are no user-facing changes here so I'm not sure we'd get a lot of feedback that's useful for the scope of this PR.

We could still discuss the interface (which isn't user-facing) or possible _future_ formats for `--about` or improvements e.g. using [`tabulate`](https://pypi.org/project/tabulate/) to generate the markdown tables.

**Comment by aaronsteers** [comment]
> @aaronsteers There are no user-facing changes here so I'm not sure we'd get a lot of feedback that's useful for the scope of this PR.

Thank you for clarifying. I think I incorrectly connected this with the implementation of:

- #1519 

> We could still discuss the interface (which isn't user-facing) or possible _future_ formats for `--about` or improvements e.g. using [`tabulate`](https://pypi.org/project/tabulate/) to generate the markdown tables.

I've not used https://pypi.org/project/tabulate/ before, but that looks like a very nice library.

File: singer_sdk/internal/about.py
Comment (aaronsteers) [inline singer_sdk/internal/about.py:1]: I'm not sure how to interpret "internal" vs the private convention of modules or code files containing the `_` prefix.

What do you think of bumping this up to the higher level and just letting the file exist as `singer_sdk/about.py`?

**Review by aaronsteers — COMMENTED** [review]
This looks really great. Thanks for creating. One question for follow-up.

Also, adding @kgpayne to get his review as well.

File: singer_sdk/internal/about.py
Comment (edgarrmondragon) [reply singer_sdk/internal/about.py:1]: Yeah, I wasn't too sure about the location either.

Comment (edgarrmondragon) [reply singer_sdk/internal/about.py:1]: https://github.com/meltano/sdk/pull/1570/commits/acc33c7698cd415bed21a0c6983a7ab74f0376e8

**Review by kgpayne — APPROVED** [review]
This is great @edgarrmondragon 🙌 My only thought is that we could in future leverage one of the markdown table libraries in place of the string manipulations. Appreciate this is a refactor though, not a rewrite of functionality, and your changes make it much easier to review now and to change if and when we want to 🙂

**Pull request merged**